Firestore/Swift/Source/SwiftAPI/Pipeline/Pipeline.swift

```diff
@@ -0,0 +1,415 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#if SWIFT_PACKAGE
+  @_exported import FirebaseFirestoreInternalWrapper
+#else
+  @_exported import FirebaseFirestoreInternal
+#endif // SWIFT_PACKAGE
+import Foundation
+
+@available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
+public struct Pipeline: @unchecked Sendable {
+  private var stages: [Stage]
+  var bridge: PipelineBridge
+  let db: Firestore
+
+  init(stages: [Stage], db: Firestore) {
+    self.stages = stages
+    self.db = db
+    bridge = PipelineBridge(stages: stages.map { $0.bridge }, db: db)
+  }
+
+  public func execute() async throws -> PipelineSnapshot {
+    return try await withCheckedThrowingContinuation { continuation in
+      self.bridge.execute { result, error in
+        if let error {
+          continuation.resume(throwing: error)
+        } else {
+          continuation.resume(returning: PipelineSnapshot(result!, pipeline: self))
+        }
+      }
+    }
+  }
+
+  /// Adds new fields to outputs from previous stages.
```

Comment:
I believe other platforms include code samples of how to use this stage. Please add them here as well.